docs/history-service.md

```diff
@@ -0,0 +1,274 @@
+- The task-execution framework itself uses a goroutine pool that is shared across shards. Ultimately, it [calls](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/common/tasks/fifo_scheduler.go#L219) `Execute` and then either `Ack` or `Nack` on each task.
+
+- Each queue processor also starts a goroutine running `processEventLoop`.
+  There are two variants of `processEventLoop`: [one for immediate queues](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/queues/queue_immediate.go#L150) such as the Transfer queue, and [one for scheduled queues](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/queues/queue_scheduled.go#L199) such as the Timer queue. For both queue types, [`checkpoint()`](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/queues/queue_base.go#L349) is called periodically, which persists queue ack levels.
+
+- [`executions` table schema in Cassandra](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/schema/cassandra/temporal/schema.cql#L7)
+- [handling](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/timer_queue_active_task_executor.go#L114) of the various types of timers in the Timer Task queue.
+
+   </details>
+   <br>
+
+#### Transfer Task queue
+
+A Transfer Task is available to be executed immediately.
+When a Transfer Task is executed, an RPC is made to the Matching Service to enqueue the Workflow Task or Activity Task specified by the Transfer Task.
+
+<details>
+<summary><i>Code entrypoints</i></summary>
+
+- The [`Execute`](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_active_task_executor.go#L114) method on a task dispatches according to task type.
+
+- For an Activity, [`processActivityTask`](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_active_task_executor.go#L169) is called, resulting in [an RPC to Matching service](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_task_executor_base.go#L121) to create the Activity Task.
+
+- Similarly, for a workflow task: [`processWorkflowTask`](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_active_task_executor.go#L217) results in an [RPC to Matching Service](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/transfer_queue_task_executor_base.go#L152) to create the Workflow Task.
+</details><br>
+
```

Comment:
So, TT is not only for RPC (for example `CloseExecutionTask`). I don't think that we need to list all cases here. These two, you mentioned, are most important, for sure. But I think it worth to say that there are other cases.